### PR TITLE
fix(6562): dashboard tests passing on iox

### PR DESCRIPTION
**Pull request by wiedld**
part of #6562 

## Reasoning:
* e2es goals:
    * prevent regressions of any existing code (by internal or external users)
    * to ensure e2e compatibility with the full stack.
       * any exceptions added are about the backend/storage
           * e.g. only tsm supports explicit bucket schema --> use cy.skipOn() as needed
           * e.g. only iox supports SQL --> use cy.skipOn() as needed
* enablement flags are selectively on:
    * which flags are required, explicitly indicates the feature coupling.
    * we could also use this to set ordering of feature deletion (when we eventually get there).
    * e.g. `showDashboardsInNewIOx` + `showVariablesInNewIOx` => dashboards have to be removed, before we can remove variables.

## Evidence of tests passing on iox:

Screenshots from after the rebase onto the proper cypress env vars.

Screen Shot 2023-02-01 at 4 47 32 PM
Screen Shot 2023-02-01 at 4 42 54 PM

Screen Shot 2023-02-01 at 5 15 50 PM
Screen Shot 2023-02-01 at 5 11 18 PM
Screen Shot 2023-02-01 at 4 59 29 PM

## Checklist
- [x] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [x] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [x] Documentation updated or issue created (provide link to issue/PR)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] Feature flagged, if applicable

### CODE REVIEW

**Review by appletreeisyellow — DISMISSED** [review]

**Review by wdoconnell — COMMENTED** [review]
Looks good - but do we need the variables flag on as well for all features? We're only flipping `showDashboardsInIOx` for a couple of these tests, it looks like.

I would also add a test along the lines of "Dashboards doesn't display if it's an IOx org."

**Comment by appletreeisyellow** [comment]
@wdoconnell I was working on `shared/dashboardsView.test.ts` and I put up a PR here https://github.com/influxdata/ui/pull/6573, which includes an additional test for "Dashboards doesn't display if it's an IOx org" case. So it's covered!

**Comment by wdoconnell** [comment]
Thanks @appletreeisyellow

**Comment by wiedld** [comment]
Had to rebase a merge conflict on 1 of the tests. (Thanks for the updates @appletreeisyellow !)

Re-ran the iox test locally, just to make sure ok
Screen Shot 2023-02-02 at 3 05 43 PM
Screen Shot 2023-02-02 at 3 13 20 PM

File: cypress/e2e/shared/dashboardsView.test.ts [hunk 1]
```diff
@@ -6,19 +6,22 @@ const isTSMOrg = !isIOxOrg
...
-          cy.visit(`${orgs}/${orgID}/dashboards-list`)
-          cy.getByTestID('tree-nav')
-        })
-      })
-    })
+    cy.flush().then(() =>
+      cy.signin().then(() =>
+        cy
+          .setFeatureFlags({
+            showDashboardsInNewIOx: true,
```
Comment (wiedld) [inline cypress/e2e/shared/dashboardsView.test.ts:13]: This specific test does not need variables in order to pass.

**Comment by wiedld** [comment]
> Looks good - but do we need the variables flag on as well for all features? We're only flipping `showDashboardsInIOx` for a couple of these tests, it looks like.
> 
> I would also add a test along the lines of "Dashboards doesn't display if it's an IOx org." And that test will be skipped if the org type is not IOx.

The tests here pass iox without variables being on. So I was thinking along the lines of keeping the minimum flags needed. But I can change that too.

**Review by appletreeisyellow — APPROVED** [review]

**Pull request merged**